mlflow/utils/autologging_utils/safety.py

```diff
@@ -834,6 +854,8 @@ def _validate(autologging_call_input, user_call_input=None):
             - for all other input types, `autologging_call_input` and `user_call_input`
               must be equivalent by reference equality or by object equality
         """
+        nonlocal autologging_integration, function_name
```

Comment:
I don't think we need `nonlocal` here since we're not mutating these vars, right?